### ISSUE TITLE
getTransactionReference() on a failed payment

### ISSUE THREAD

**Issue by adamlc** [thephpleague/omnipay-stripe#7]
I only use Omnipay for Stripe at the moment but I've noticed that when a transaction has failed $response->getTransactionReference() returns null, obviously you'd expect it to always return the transaction id regardless.

The transaction id is available in the response from Stripe. I'm currently getting it by doing the following:

```php
$transData = $response->getData();

if (isset($transData['error']['charge'])) {

    return $transData['error']['charge'];

 }
```

Is this is expected behaviour? If it is then should it be changed? What are your thoughts?

**Comment by delatbabel** [comment]
Hrm. In the general case, most gateways don't return a transaction ID when the transaction has failed. Stripe is the odd case here. So in general I would expect getTransactionReference() to return null or empty when the transaction fails.  Having said that, it would be a simple matter to alter the Response class to return the value from [error][charge] in the case of a failed transaction if you really need to store transaction IDs for failed transactions. Be aware that it probably won't work for other gateways, though.

**Comment by delatbabel** [comment]
This should be fixed in PR #13

**Comment by NoelDavies** [comment]
Woo, you have just added what I was going to create  PR for
